src/Heatmap/HeatMapInterpolationPoints.swift

```diff
@@ -0,0 +1,399 @@
+/* Copyright (c) 2020 Google Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import Foundation
+import GooglePlaces
+import GoogleMaps
+import GoogleMapsUtils
+
+/// This class will create artificial points in surrounding locations with appropriate intensities interpolated by neighboring intensity values.
+class HeatMapInterpolationPoints {
+    
+    /// The input data set; each entry contains the coordinates and the 
+    private var data = [[Double]]()
+    
+    /// The heat map layer
+    private let heatMapLayer: GMUHeatmapTileLayer = GMUHeatmapTileLayer()
+    
+    /// The heat map colors and gradient
+    private var gradientColors = [UIColor.green, UIColor.red]
+    private var gradientStartPoints = [0.005, 0.7] as [NSNumber]
+    
+    /// The user should be able to access thisdata as well as the size of the data
+    public var heatMapPoints = [GMUWeightedLatLng]()
+    
+    // MARK: Functions that parse given data needed to build an interpolated heat map from
+    
+    /// Takes in the data set file and calculates the weights of the given points; these values are used for interpolation later
+    ///
+    /// - Parameter file: The input data set file; it must have a json extension.
+    public func setData(file: String) {
```

Comment:
I'm not a fan of providing this API at it imposes a certain structure on a file that might not be as common. For example, what if the lat, lng values are somewhere nested? This is a similar reason why the heatmap class does not support this. See in our docs how reading from file is not handled by the class: https://developers.google.com/maps/documentation/ios-sdk/utility/heatmap

Reply:
Oh ok. Is it ok for the interpolator to take in a set of data points then? As in, let the developer read an input file elsewhere and then pass in that data to the interpolator?